src/validator_main.cpp

```diff
@@ -160,15 +160,20 @@ int main(int argc, char** argv)
 
     po::options_description desc = build_command_line_options();
     po::variables_map vm;
-    po::store(po::parse_command_line(argc, argv, desc), vm);
-    po::notify(vm);
+    po::parsed_options parsed = po::command_line_parser(argc, argv).options(desc).allow_unregistered().run();
```

Comment:
what about extracting this behaviour to a function `build_variables_map`? including the check for unused args